Adds migration guide and changes for vendor proxies (#712)
piwik/developer-documentation: docs/5.x/piwik-on-the-command-line.md

This will create a folder named `Commands` within your plugin along with a PHP file which represents the actual command. You can add an unlimited number of commands to a plugin.

To learn how to flesh out your command, you can read the [Symfony Console documentation](https://symfony.com/doc/current/components/console/index.html).

Nevertheless, there are some differences when building Matomo commands. As plugins should avoid any direct references to Matomo's dependencies we have our own Command class to inherit from.
This class inherits from the Symfony Console Command class, but has some adjustments that will avoid the direct requirement of using typehints like `InputInterface` or `OutputInterface`, class constants from `InputOption` or `InputArgument` and any Console helpers.

> * Methods like `run`, `execute`, `interact` or `initialize` can not be overwritten. Instead, use our custom methods prefixed with `do`: `doExecute`, `doInteract` or `doInitialize`

Reviewer: ```suggestion
* Methods like `run`, `execute`, `interact` or `initialize` cannot be overwritten. Instead, use our custom methods prefixed with `do`: `doExecute`, `doInteract` or `doInitialize`
```

> * Where ever you need to work with input or output use `$this->getInput()` or `$this->getOutput()` instead. Don't use `InputInterface` or `OutputInterface` as method typehints.

Reviewer: ```suggestion
* Wherever you need to work with input or output, use `$this->getInput()` or `$this->getOutput()` instead. Don't use `InputInterface` or `OutputInterface` as method typehints.
```

> * When configuring input options and arguments `addOption` and `addArgument` can not be used

Reviewer: ```suggestion
* When configuring input options and arguments `addOption` and `addArgument` cannot be used
```

  * For arguments use `addOptionalArgument` or `addRequiredArgument`
  * For options use `addNegatableOption`, `addOptionalValueOption`, `addNoValueOption` or `addRequiredValueOption`

> * Directly using any console helpers is now prohibited

Reviewer: ```suggestion
* Directly using any of the console helpers is now prohibited
```

>   * When needing user input use the new methods `askForConfirmation`, `askAndValidate` or `ask`

Reviewer: ```suggestion
  * To get user's input use the new methods `askForConfirmation`, `askAndValidate` or `ask`
```

  * For progress bars use the methods `initProgressBar`, `startProgressBar`, `advanceProgressBar` and `finishProgressBar`

>   * Tables can be rendered using the new method `renderTable`

Reviewer: ```suggestion
  * To render tables use the new method `renderTable`
```
